Source/Evolve/WarpXEvolve.cpp

```diff
@@ -217,8 +217,16 @@ WarpX::Evolve (int numsteps)
             // B : guard cells are NOT up-to-date
         }
 
-        if (cur_time + dt[0] >= stop_time - 1.e-3*dt[0] || step == numsteps_max-1) {
+        if ((cur_time + dt[0] >= stop_time - 1.e-3*dt[0] || step == numsteps_max-1) &&
```

Comment:
This is a bit long. Maybe a temporary bool could simplify the logic in the `if`?

Reply:
Thanks for the comments! Do you have a suggestion of how to simplify this?